gson/src/main/java/com/google/gson/MissingFieldValueStrategy.java

```diff
@@ -0,0 +1,77 @@
+package com.google.gson;
+
+import com.google.gson.internal.reflect.ReflectionHelper;
+import com.google.gson.reflect.TypeToken;
+import java.lang.reflect.Field;
+
+/**
+ * A strategy defining how to handle missing field values during reflection-based deserialization.
+ *
```

Comment:
Maybe this should have a warning that users should be careful when implementing security critical validation with this, and add extensive tests. Because if they forget to call `setMissingFieldValueStrategy` or use a default `Gson` instance somewhere, then their custom strategy will not be used.